### PR TITLE
Allow configuring Public Access Block for S3 buckets

### PR DESCRIPTION
**Pull request by ronnyroeller**
Amazon S3 allows to block public access on bucket level. This prevents developers from putting a ACL that would allow e.g. upload of public files.

See https://docs.aws.amazon.com/AmazonS3/latest/dev/access-control-block-public-access.html

### CODE REVIEW

File: src/resources/s3.js [hunk 1]
```diff
@@ -77,6 +77,29 @@ class S3Bucket { // eslint-disable-line padded-blocks
...
 
+	/**
+	 * Translate the logging configuration into the 'logging params' for the AWS SDK.
+	 *
+	 * @param {string} bucketName the bucket name
+	 * @param {PublicAccessBlockConfiguration} publicAccessBlockConfiguration Public Access Block configuration
+	 * @returns {Object} the parameters for `putPublicAccessBlock`
+	 */
+	_translatePublicAccessBlockConfiguration(bucketName, publicAccessBlockConfiguration) {
+		let status;
```
Comment (ankon) [inline src/resources/s3.js]: `status` is a weird name here.

File: src/resources/s3.js [hunk 2]
```diff
@@ -77,6 +77,29 @@ class S3Bucket { // eslint-disable-line padded-blocks
...
+		if (!publicAccessBlockConfiguration) {
+			status = {};
+		} else {
+			status = {
+				BlockPublicAcls: publicAccessBlockConfiguration.blockPublicAcls,
```
Comment (ankon) [inline src/resources/s3.js]: Could you also add all other properties of https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-s3-bucket-publicaccessblockconfiguration.html please?

File: src/resources/s3.js [hunk 3]
```diff
@@ -77,6 +77,29 @@ class S3Bucket { // eslint-disable-line padded-blocks
 		};
 	}
 
+	/**
+	 * Translate the logging configuration into the 'logging params' for the AWS SDK.
+	 *
+	 * @param {string} bucketName the bucket name
+	 * @param {PublicAccessBlockConfiguration} publicAccessBlockConfiguration Public Access Block configuration
```
Comment (ankon) [inline src/resources/s3.js]: Please document the type (see for example `LoggingConfiguration`)

File: src/resources/s3.js (hunk 1 above)
Comment (ronnyroeller) [reply src/resources/s3.js]: Addresses by https://github.com/Collaborne/kubernetes-aws-resource-service/pull/43/commits/93bd84280243aff0ac79991da6a749b97034808c

File: src/resources/s3.js (hunk 2 above)
Comment (ronnyroeller) [reply src/resources/s3.js]: Fixed by https://github.com/Collaborne/kubernetes-aws-resource-service/pull/43/commits/93bd84280243aff0ac79991da6a749b97034808c

File: src/resources/s3.js (hunk 3 above)
Comment (ronnyroeller) [reply src/resources/s3.js]: Fixed by https://github.com/Collaborne/kubernetes-aws-resource-service/pull/43/commits/2e686d454a0b5b5bee247c946f156700a869eb8c

**Comment by ronnyroeller** [comment]
@ankon All raised points have been addresses (in separated commits for easier review).
Could you please have another look?

**Comment by ankon** [comment]
Thanks, looks good!

Will leave this running in our test environment for a while, and then make a release.

**Pull request merged**